### PR TITLE
Try fix image block

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Alternative to #21160. This is a more complete fix, I think.

Fixes clicking on the block when it's floated.
Fixes centre alignment.
Fixes toolbar position when floated.
Fixes block outline when floated.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/image/edit.js [hunk 1]
```diff
@@ -509,17 +509,19 @@ export class ImageEdit extends Component {
...
 		/* eslint-disable jsx-a11y/click-events-have-key-events */
 		return (
 			<>
 				{ controls }
-				<AlignmentWrapper>
-					<BlockContentWrapper className={ classes }>
+				<div
+					className={
+						needsAlignmentWrapper
+							? 'wp-block block-editor-block-list__block'
```
Comment (ellatrix) [inline packages/block-library/src/image/edit.js]: Ideally these classes are not needed, and ideally, we provide an alignment wrapper component that the block can wrap around the block.

Comment (youknowriad) [reply packages/block-library/src/image/edit.js]: Why these are needed?

Comment (youknowriad) [reply packages/block-library/src/image/edit.js]: I guess it's another argument to absorb the alignment in `Block.*` for now.

Comment (youknowriad) [reply packages/block-library/src/image/edit.js]: I'm fine with this for now (with a comment) while we continue the work to improve alignments

Comment (ellatrix) [reply packages/block-library/src/image/edit.js]: It's needed so the div has the proper width and margins. Yes, we'd need to comment and continue with figuring out how to best proceed with the alignment issue.

**Comment by ellatrix** [comment]
This needs a thorough test since we'll need to do a patch release for Gutenberg 7.8 with this change.

**Review by youknowriad — APPROVED** [review]
I tested this and it seems to work well in 2020

Some alignments are broken in 2019 but I believe that it's broken in master too.

**Comment by jasmussen** [comment]
This tests well for me also. Both with no editor style and TwentyTwenty. TwentyNineteen, as Riad mentions, has some separate issues. 

I also confirmed that very very long multi-line captions still work as they should. Nice.

The following is unrelated, but something we really should look at: there's no right margin on a left floated image, or left margin on a right floated image:

Screenshot 2020-03-26 at 14 28 08

**Comment by ellatrix** [comment]
> The following is unrelated, but something we really should look at: there's no right margin on a left floated image, or left margin on a right floated image:

That seems to be the case only if the image is resized 🤔

**Comment by jasmussen** [comment]
Interesting.

**Comment by jasmussen** [comment]
Possibly a result of this PR combined with the recent refactor of the main canvas that removed the permalink, but now fullwide images aren't fullwide:

Screenshot 2020-03-27 at 15 11 36

I'll take a look.

**Comment by ellatrix** [comment]
@jasmussen Thanks for looking into it

**Comment by ellatrix** [comment]
I guess this needs to be fixed before we can do a patch release.

**Comment by jasmussen** [comment]
I fixed it in https://github.com/WordPress/gutenberg/pull/21201 but I'm not sure how to proceed with the feedback.

**Comment by strarsis** [comment]
Does Gutenberg use automated end to end tests for these kinds of bugs?
Would a test catch this issue (unable to select an image block in certain cases) the next time?

**Comment by dnnsjsk** [comment]
> Possibly a result of this PR combined with the recent refactor of the main canvas that removed the permalink, but now fullwide images aren't fullwide:
> 
> Screenshot 2020-03-27 at 15 11 36
> 
> I'll take a look.

I'm having the same issue since 7.8.1. Also alignment are all over the shop, I reverted back to 7.8 for now.

**Pull request merged**